iroh-net/src/magic_endpoint.rs

```diff
@@ -78,14 +78,31 @@ impl From<(PublicKey, Option<Url>, &[SocketAddr])> for NodeAddr {
 }
 
 /// Addressing information to connect to a peer.
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Default)]
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Default)]
 pub struct AddrInfo {
     /// The peer's home DERP url.
     pub derp_url: Option<Url>,
     /// Socket addresses where the peer might be reached directly.
     pub direct_addresses: BTreeSet<SocketAddr>,
 }
 
+impl Debug for AddrInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("AddrInfo")
+            .field("derp_url", &self.derp_url.as_ref().map(DD))
+            .field("direct_addresses", &self.direct_addresses)
+            .finish()
+    }
+}
+
+struct DD<T: std::fmt::Display>(T);
```

Comment:
not really a need for a new type here, you can directly format the field instead (we already do this somewhere else as well, maybe somewhere in netcheck, because i recall doing exactly the same to a URL before)